### PR TITLE
refactor(maven): Unified result type for http fetch

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/maven/util.ts [hunk 1]
```diff
@@ -312,25 +333,21 @@ export async function downloadMavenXml(
...
-        xml: new XmlDocument(body),
-        isCacheable: !res.authorization,
-      };
-    }
+    const rawResult = await downloadHttpProtocol(http, pkgUrl);
+    const xmlResult = rawResult.transform(({ isCacheable, data }): MavenXml => {
+      const xml = new XmlDocument(data);
+      return { isCacheable, xml };
+    });
+    return xmlResult.unwrapOr({});
```
Comment (viceice) [inline lib/modules/datasource/maven/util.ts:341]: does it no longer throw if invalid XML?

Comment (viceice) [reply lib/modules/datasource/maven/util.ts:341]: can you please log a message for invalid XML. eg if a user configured something wrong for the registry url?

**Review by viceice — COMMENTED** [review]
see comment

File: lib/modules/datasource/maven/util.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/datasource/maven/util.ts:341]: I'll have better handling of it it in further PR

Comment (zharinov) [reply lib/modules/datasource/maven/util.ts:341]: Could we live about a week without XML handling?

**Review by viceice — APPROVED** [review]
please add XML parse error logging as soon as possible

**Pull request merged**